crates/ibc/src/core/context.rs

```diff
@@ -292,8 +292,19 @@ pub trait ValidationContext: Router {
     /// Returns the ConnectionEnd for the given identifier `conn_id`.
     fn connection_end(&self, conn_id: &ConnectionId) -> Result<ConnectionEnd, ContextError>;
 
-    /// Validates the `ClientState` of the client on the counterparty chain.
-    fn validate_self_client(&self, counterparty_client_state: Any) -> Result<(), ConnectionError>;
+    /// Validates the client state of the host chain that is stored on the
+    /// counterparty chain.
```

Comment:
```suggestion
    /// Validates the `ClientState` of the client (a client referring to host) stored on the counterparty chain against the host's internal state.
```
Tried a different wording; I read your sentence as "the client state of the host". Hopefully this new wording is a step in the right direction; feel free to reword once again.